segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/Ledgers.java

```diff
@@ -35,13 +41,27 @@
     static final String PROPERTY_VALUE_APPLICATION = "Pravega";
     static final String PROPERTY_LOG_ID = "BookKeeperLogId";
     static final int NO_LOG_ID = -1;
-    static final long NO_LEDGER_ID = LedgerHandle.INVALID_LEDGER_ID;
-    static final long NO_ENTRY_ID = LedgerHandle.INVALID_ENTRY_ID;
+    static final long NO_LEDGER_ID = -1; // LedgerHandle.INVALID_LEDGER_ID
+    static final long NO_ENTRY_ID = -1; // LedgerHandle.INVALID_ENTRY_ID
     /**
      * How many ledgers to fence out (from the end of the list) when acquiring lock.
      */
     static final int MIN_FENCE_LEDGER_COUNT = 2;
 
+    /**
+     * Converts exceptions to BKException. Fallback to BKUnexpectedConditionException
+     * for non BookKeeper exceptions.
+     */
+    private static final Function<Throwable, BKException> BK_EXCEPTION_HANDLER = cause -> {
+        if (cause instanceof BKException) {
+            return (BKException) cause;
+        } else {
+            BKUnexpectedConditionException wrapper = new BKUnexpectedConditionException();
```

Comment:
According to codecov, this path is not being exercised. Could we test this handler directly to exercise all code paths?

Reply:
sure, will do

Reply:
done